lib/engine/round/g_1836_jr_30/operating.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require_relative '../operating'
+
+module Engine
+  module Round
+    module G1836Jr30
+      class Operating < Operating
+        def initialize(entities, game:, round_num: 1, **opts)
+          super
+          @depot_trains_bought = []
+        end
+
+        def start_operating
+          @depot_trains_bought = []
+          super
+        end
+
+        def buyable_trains
+          super.reject { |x| x.from_depot? && @depot_trains_bought.include?(x.distance) }
+        end
+
+        def can_buy_train?
+          super && (@depot_trains_bought.empty? || buyable_trains.any?)
+        end
+
+        def process_buy_train(action)
+          # Since the train won't be in the depot after being bougth store the state now.
+          add_to_list = action.train.from_depot?
+          super
```

Comment:
can't we do

@depoat_trains_bought << action.train.sym if action.train.from_depot?
super

Reply:
nope, since process_buy_trains calls buyable_trains and means no train can be bought